tests/test_session_builder.c

```diff
@@ -246,14 +241,15 @@ START_TEST(test_basic_pre_key_v3)
     result = curve_generate_key_pair(global_context, &bob_pre_key_pair);
     ck_assert_int_eq(result, 0);
 
-    ec_key_pair *bob_signed_pre_key_pair = 0;
-    result = curve_generate_key_pair(global_context, &bob_signed_pre_key_pair);
-    ck_assert_int_eq(result, 0);
-
     ratchet_identity_key_pair *bob_identity_key_pair = 0;
     result = signal_protocol_identity_get_key_pair(bob_store, &bob_identity_key_pair);
     ck_assert_int_eq(result, 0);
 
+    session_signed_pre_key *bob_signed_pre_key_record = 0;
+    result = signal_protocol_key_helper_generate_signed_pre_key(&bob_signed_pre_key_record, bob_identity_key_pair, 22, time(0), global_context);
+    ck_assert_int_eq(result, 0);
+    ec_key_pair *bob_signed_pre_key_pair = session_signed_pre_key_get_key_pair(bob_signed_pre_key_record);
+
```

Comment:
Looks good

Reply:
![image](https://user-images.githubusercontent.com/7421424/77019001-c49d4800-6955-11ea-8c4a-433d2e0aa085.png)

- The first `passed` log is from `test_schnorr_verification`

- The `[WARNING] no signed pre key!` log is from `test_basic_pre_key_v2` and it is expected because the test checks that processing the pre key bundle fails when there is a missing signed pre key

- The `Pre-interaction tests complete`...`Post-interaction tests complete` logs are from `test_basic_pre_key_v3` and they are expected (no warnings/errors in any of those messages)

- The remaining logs are from `test_bad_signed_pre_key_signature`. You'll see `passed` followed by `[WARNING] invalid signature on device key!` but you should just make sure that the last log is not `[WARNING] invalid signature on device key!`. This is the expected behavior because the unit test repeatedly processes the pre key bundle with a corrupted signature — the schnorr proof passes but the signature verification is meant to fail with an invalid signature error. The unit test then passes the correct signature and that is why the last log from the unit test should not be `[WARNING] invalid signature on device key!`

Reply:
With that said, to answer your question, yes, this is expected behavior

Reply:
Awesome, thanks for clarifying!